### PR TITLE
fix: FreeBSD build problems

### PR DESCRIPTION
**Pull request by ethan-thompson**
Check if system is running BSD. If so, update declared C types accordingly.

I had to make these changes to get the package to build and install on FreeBSD 14.1. Creating this PR to save others the headache.

### CODE REVIEW

**Comment by whooo** [comment]
Could you please sign off the commit to agree with the DCO

**Comment by ethan-thompson** [comment]
Ah, yes, sorry about that!

**Comment by whooo** [comment]
There where some changes in a upstream dependency that broke the CI, but it's fixed on master now, could you rebase on master and push?

File: setup.py [hunk 1]
```diff
@@ -183,9 +184,20 @@ def get_mappings(self):
...
-            header_path, cpp_args=["-D__extension__=", "-D__attribute__(x)="]
-        )
+
+        if platform.system() == "FreeBSD":
+            pdata = preprocess_file(
+                header_path,
+                cpp_args=[
+                    "-D__builtin_va_list=char*",
+                    "-D__extension__=",
+                    "-D__attribute__(x)="
```
Comment (whooo) [inline setup.py:194]: Please add a `,` at the end

**Comment by whooo** [comment]
Last nitpick, could squash the commits into one? Will make it easier to follow the commits later on if needed

**Comment by ethan-thompson** [comment]
No problem!

**Review by whooo — APPROVED** [review]

**Comment by whooo** [comment]
thanks for your contribution!

**Pull request merged**